lib/pure/asyncmacro.nim

```diff
@@ -472,27 +472,16 @@ proc stripAwait(node: NimNode): NimNode =
 proc splitParamType(paramType: NimNode, async: bool): NimNode =
   result = paramType
   if paramType.kind == nnkInfix and $paramType[0].ident in ["|", "or"]:
-    let firstType = paramType[1]
-    let firstTypeName = $firstType.ident
-    let secondType = paramType[2]
-    let secondTypeName = $secondType.ident
+    let firstAsync = "async" in ($paramType[1].ident).normalize
+    let secondAsync = "async" in ($paramType[2].ident).normalize
 
     # Make sure that at least one has the name `async`, otherwise we shouldn't
     # touch it.
-    if not ("async" in firstTypeName.normalize or
```

Comment:
I don't think this should be removed.

Reply:
New version does the same. If both types names do not contain "async" substring, the function returns `paramType`.

Reply:
But yes, I forgot to remove the comment